driver-core/src/main/java/com/datastax/driver/core/QueryLogger.java

```diff
@@ -0,0 +1,544 @@
+         * The default for this setting is {@link #DEFAULT_SLOW_QUERY_THRESHOLD_MS}.
+         *
+         * @param slowQueryLatencyThresholdMillis Slow queries threshold in milliseconds.
+         *                                        It must be strictly positive.
+         * @return this {@link Builder} instance (for method chaining).
+         * @throws IllegalArgumentException if {@code slowQueryLatencyThresholdMillis <= 0}.
+         */
+        public Builder withSlowQueryLatencyThresholdMillis(long slowQueryLatencyThresholdMillis) {
+            instance.setSlowQueryLatencyThresholdMillis(slowQueryLatencyThresholdMillis);
+            return this;
+        }
+
+        /**
+         * Set the maximum length of a CQL query string that can be logged verbatim
+         * by the driver. Query strings longer than this value will be truncated
+         * when logged.
+         * The default for this setting is {@link #DEFAULT_MAX_QUERY_STRING_LENGTH}.
+         *
+         * @param maxQueryStringLength The maximum length of a CQL query string
+         *                             that can be logged verbatim by the driver.
+         *                             It must be strictly positive or {@code -1},
+         *                             in which case the query is never truncated
+         *                             (use with care).
+         * @return this {@link Builder} instance (for method chaining).
+         * @throws IllegalArgumentException if {@code maxQueryStringLength <= 0 && maxQueryStringLength != -1}.
+         */
+        public Builder withMaxQueryStringLength(int maxQueryStringLength) {
+            instance.setMaxQueryStringLength(maxQueryStringLength);
+            return this;
+        }
+
+        /**
+         * Set the maximum length of a query parameter value that can be logged verbatim
+         * by the driver. Parameter values longer than this value will be truncated
+         * when logged.
+         * The default for this setting is {@link #DEFAULT_MAX_PARAMETER_VALUE_LENGTH}.
+         *
+         * @param maxParameterValueLength The maximum length of a query parameter value
+         *                                that can be logged verbatim by the driver.
+         *                                It must be strictly positive or {@code -1},
+         *                                in which case the parameter value is never truncated
+         *                                (use with care).
+         * @return this {@link Builder} instance (for method chaining).
+         * @throws IllegalArgumentException if {@code maxParameterValueLength <= 0 && maxParameterValueLength != -1}.
+         */
+        public Builder withMaxParameterValueLength(int maxParameterValueLength) {
+            instance.setMaxParameterValueLength(maxParameterValueLength);
+            return this;
+        }
+
+        /**
+         * Set the maximum length of a query parameter value that can be logged verbatim
```

Comment:
This sentence is the same as the one in withMaxParameterValueLength, shouldn't this be "Set the maximum number of query parameters that can be logged by the driver" similar to the param argument below?